module/VuFind/src/VuFind/DigitalContent/OverdriveConnector.php

```diff
@@ -276,10 +276,12 @@ public function getAvailability($overDriveId)
                 }
             } else {
                 $result->status = true;
-                $res->copiesAvailable ??= 0;
-                $res->copiesOwned ??= 0;
-                $res->numberOfHolds ??= 0;
-                $res->code = 'od_none';
+                if ($res) {
+                    $res->copiesAvailable ??= 0;
+                    $res->copiesOwned ??= 0;
+                    $res->numberOfHolds ??= 0;
+                    $res->code = 'od_none';
+                }
```

Comment:
This has the effect of giving incorrect information in the UI (saying 0/0 copies are available) regardless of the truth.  That is better than the ugly error output that happens without the `if ($res)` check, but still not ideal.

Reply:
I think the question here is, what is the truth in the situation where we have a response but these values are missing? In theory, this is probably never supposed to happen, so backfilling with zeroes is mainly intended to protect against undefined/unexpected situations. In our case, this just helps us with our local simulated data, which is incomplete. If there's a real-world situation where OverDrive returns incomplete data, and what that means is something other than "the value is zero," I'm certainly open to adding some more specialized error handling. But in the meantime, I'm not sure what would be the best thing to do, so I'm inclined not to try. :-)